src/analysis/normed/group/basic.lean

```diff
@@ -1689,6 +1689,37 @@ instance pi.normed_comm_group [Π i, normed_comm_group (π i)] : normed_comm_gro
 
 end pi
 
+/-! ### Multiplicative opposite -/
+
+namespace mul_opposite
+
+instance [has_norm E] : has_norm (Eᵐᵒᵖ) :=
+{ norm := λ x, ‖x.unop‖ }
```

Comment:
Is this instance likely to cause trouble on `normed_group`s (ie, the multiplicative case rather than the additive case?)

Reply:
Why would it do that? And why not use `to_additive` throughout this section?

Reply:
No I don't see how it could cause problems. Opposites of normed groups don't inherit a different norm in the literature. Instead, we have a few type synonyms that do get endowed with different/new norms.

Reply:
> And why not use `to_additive` throughout this section?

That doesn't work here, because there is both an addition and a multiplicative version, and the to_additive version would have them swapped.

Reply:
> Why would it do that?

Because I think this isn't true:

```lean
instance [seminormed_group E] : seminormed_group Eᵐᵒᵖ :=
{ dist_eq := mul_opposite.rec $ λ x, mul_opposite.rec $ λ y, begin
    rw [←norm_unop, unop_div, unop_op, unop_op, dist_op, dist_eq_norm_div, div_eq_mul_inv],
    sorry -- `‖x * y⁻¹‖ = ‖y⁻¹ * x‖`
  end,
  to_pseudo_metric_space := by apply_instance }
```

Reply:
Phrased another way, it needs `dist x y = dist x⁻¹ y⁻¹` which is not true in general (except in commutative groups where `mul_opposite` is boring)

Reply:
> > And why not use `to_additive` throughout this section?
> 
> That doesn't work here, because there is both an addition and a multiplicative version, and the to_additive version would have them swapped.

That doesn't apply to the general `has_norm`, though.

Reply:
Aha, yes, I wasn't thinking carefully.

Reply:
> That doesn't apply to the general `has_norm`, though.

Indeed, but I've removed that for now since the utility is dubious, and added a comment explaining that opinion.